group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -983,27 +1052,125 @@ private CoordinatorResult<ConsumerGroupHeartbeatResponseData, Record> consumerGr
         return new CoordinatorResult<>(records, response);
     }
 
+    private void replaceStaticMemberInConsumerGroup(
+        String groupId,
+        List<Record> records,
+        ConsumerGroup group,
+        ConsumerGroupMember member,
+        ConsumerGroupMember existingStaticMember,
+        ConsumerGroupMember updatedMember
+    ) {
+        // Write tombstones for the departed static member. Follow the same order as the one followed when fencing
+        // a member
+        records.add(newCurrentAssignmentTombstoneRecord(group.groupId(), existingStaticMember.memberId()));
+        records.add(newTargetAssignmentTombstoneRecord(group.groupId(), existingStaticMember.memberId()));
+        records.add(newMemberSubscriptionTombstoneRecord(group.groupId(), existingStaticMember.memberId()));
```

Comment:
We have similar code somewhere else. Could we add a method for this and reuse it?

Reply:
Done.